### ISSUE TITLE
Warn about redundant 'return None' statements

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#3658]
This may annoy the hell out of people, but here goes. Python functions that fall off the end of the control flow return `None`. Therefore, if the final line of a function is `return None`, it is redundant. Pylint could emit a warning suggesting the removal of the final line.

**Comment by bje-** [comment]
This is probably more trouble than it's worth. If there is more than one path to the function exit, removing the last `return None` can trigger this warning:

```
Either all return statements in a function should return an expression, or none of them should.
```
